### PR TITLE
feat(server): Read default configs from environment variables

### PR DESCRIPTION
**Pull request by apocelipes**
BREAKING CHANGE: worker数修改后需要重新评估调整服务的资源用量。

### CODE REVIEW

File: sea/app.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ def __init__(self, root_path, env):
         self.name = os.path.basename(root_path)
         self.env = env
         self.config = self.config_class(root_path, self.default_config)
+        # only filter default configurations
+        self.config.filter_env()
```
Comment (fenngwd) [inline sea/app.py:46]: move to https://github.com/shanbay/sea/blob/master/sea/__init__.py#L31
@apocelipes

File: sea/config.py [hunk 2]
```diff
@@ -45,5 +47,23 @@ def get_namespace(self, namespace, lowercase=True, trim_namespace=True):
             rv[key] = v
         return ConstantsObject(rv)
 
+    def filter_env(self):
```
Comment (fenngwd) [inline sea/config.py:50]: rename to load_config_from_env

File: sea/config.py [hunk 3]
```diff
@@ -45,5 +47,23 @@ def get_namespace(self, namespace, lowercase=True, trim_namespace=True):
...
+        """
+        keys = self.keys()
+
+        for k in keys:
+            env_value = os.getenv(k)
+            if not env_value:
+                continue
+            value_type = type(self[k])
+            if value_type is bool:
+                self[k] = env_value == 'True'
```
Comment (fenngwd) [inline sea/config.py:64]: replace function with disutils.util.strtobol

File: sea/config.py [hunk 4]
```diff
@@ -45,5 +47,23 @@ def get_namespace(self, namespace, lowercase=True, trim_namespace=True):
...
+            if value_type is str or value_type is int:
+                self[k] = value_type(env_value)
```
Comment (fenngwd) [inline sea/config.py:66]: check number type by : `isinstance(x, numbers.Number)`

File: tests/test_config.py [hunk 5]
```diff
@@ -29,6 +30,26 @@ class Test(Base):
     assert '<Config' in s
 
 
+def test_config_filter_env():
+    os.environ['KEY_INT'] = "300"
```
Comment (fenngwd) [inline tests/test_config.py:34]: testing overriding logic

**Review by fenngwd — CHANGES_REQUESTED** [review]

File: sea/config.py (hunk 3 above)
Comment (apocelipes) [reply sea/config.py:64]: Package disutils has been marked as deprecated since 3.10 and will be removed from 3.12. I copied it to sea.utils.

**Pull request merged**